cmake/Rust.cmake

```diff
@@ -0,0 +1,48 @@
+include(FetchContent)
```

Comment:
This is going to make builds using OBS difficult; is it possible to include Corrosion in the tree and use it as a subdirectory instead?

Reply:
> Also, distro packagers are not going to like this, but it's fine if you try `find_package` first:
> 
> ```
> find_package(Corrosion)
> if (NOT Corrosion_FOUND)
>    include(FetchContent)
>    ...
> endif()
> ```

Distro packager for Arch Linux here, this would make our lives so much easier with the ability to use system libraries.

Reply:
I'm happy to add that, but my hope is that no distro packages fish while it uses Corrosion. This use is meant to be temporary during a single development cycle; by the next release I hope to have no CMake at all, and therefore no Corrosion.

Reply:
> This is going to make builds using OBS difficult; is it possible to include Corrosion in the tree and use it as a subdirectory instead?

Corrosion can be added as a subdirectory/ submodule.

Reply:
Hello, throwing myself into this conversation with some additional info that distro maintainers might find useful. `FetchContent` has always supported placing the source directory for any dependency in any location and then setting `-DFETCHCONTENT_SOURCE_DIR_<uppercase-name>=<path>`.

In this case, this would allow package maintainers to do `-DFETCHCONTENT_SOURCE_DIR_CORROSION=<path/to/corrosion/sources>` and not interrupt the workflow. No conditional `find_package` call is required.

Additionally, [starting in CMake 3.24](https://cmake.org/cmake/help/latest/guide/using-dependencies/index.html#fetchcontent-and-find-package-integration), it's possible to tell `FetchContent_MakeAvailable` to call `find_package` first (and vice versa where `find_package` calls `FetchContent_MakeAvailable` if no file was found).

Reply:
Thank you, this is helpful. For Corrosion, my preferences are (in order): use FetchContent as in the PR (perhaps using bruxisma's tips), or add Corrosion as a fish-shell submodule, or directly check in Corrosion. Not sure what's best for OBS.